packages/crud/src/vaadin-crud-grid.js

```diff
@@ -128,27 +146,60 @@ class CrudGrid extends IncludedMixin(Grid) {
       .replace(/^./, (match) => match.toUpperCase());
   }
 
+  /** @private */
+  __getGroupDepth(column) {
+    if (column === this) {
+      return 0;
+    } else {
+      return 1 + this.__getGroupDepth(column.parentElement);
+    }
+  }
+
   /** @private */
   __createColumn(parent, path) {
-    const col = document.createElement('vaadin-grid-column');
+    const parentGroupDepth = this.__getGroupDepth(parent);
+    // Deduct 1 from the item property depth to account for the value
+    if (parentGroupDepth < this.__itemPropertyDepth - 1) {
```

Comment:
Is there a necessity for creating additional groups when `this.noSort = true` and `this.noFilter = true`? 

If however decide to ensure groups for every column to match the items' depth, I would suggest that we moved this logic to the `__createColumns` method which already takes a part in creating the tree.

Reply:
Moved the logic to `__createColumns` which is in charge of constructing the column structure, as suggested. Did some necessary refactoring & documentation for the function.